### DISCUSSION TITLE
Android Performance

### DISCUSSION THREAD

**Discussion by c068162** [bepu/bepuphysics2#320] [Q&A]
Hello, I'm making an Android game using v1 version in Unity, I'm having serious performance issues I'm using a network communication model with prediction and rollback, I lowered the update interval of Space to 1/30, it's on PC The performance is very good, usually updating Space four times per frame, and the time is only 1ms. However, when I used Android to test, its performance fluctuated greatly, and the time reached 10ms. Did I do something wrong? I am working on 2.30 Version, will upgrading to v2.30 improve the performance of the mobile terminal?

**Comment by RossNordby** [comment]
That's pretty hard to diagnose; there are a lot of unknown variables in play. If by v2.30 you mean bepuphysics2 2.3.X, bepuphysics2 will likely not work well in Unity at all until they move to the [newer runtimes](https://forum.unity.com/threads/unity-future-net-development-status.1092205/).

**Reply by c068162** [reply]
![04318992644248f6ce30b4b0cc7ad50](https://github.com/bepu/bepuphysics2/assets/121477712/26ac84ce-bef0-4838-82a6-eec757313f50)
My character is a complex, consisting of four Boxes, and the performance fluctuations are very obvious when the character collides with the wall, is there a way to make the performance fluctuations smoother，When I update Space four times in a logical frame, the performance cost of each update decreases from 1.5ms to 0.8ms (two time steps for each update)

**Reply by RossNordby** [reply]
An increase in cost is expected if there are more collision tests are being performed, but it's very hard to say whether the increase is reasonable in context, and it's also hard to say what to do about it if it's coming from runtime-related stuff. If you can't observe similar fluctuations on other platforms (in terms of relative scale), then it's more likely to be some weird runtime specific behavior that I can't really help with.

Updating the physics in a tight loop may go faster due to having the whole simulation in cache after the first one.

**Reply by c068162** [reply]
Well, thanks for your reply, I lowered the configuration of Space and limited the number of frames predicted, and now I find that it can run stably on mobile platforms, the unstable factor is network fluctuations, resulting in some logical frames that will not handle rollbacks, and it seems that the game framework that predicts rollbacks is not very suitable for running on mobile platforms.
